gcp/client.py

```diff
@@ -1,81 +1,160 @@
+import warnings
+
 from apiclient.discovery import build as build_service
+from apiclient.errors import HttpError
+
+warnings.filterwarnings(
+    "ignore", "Your application has authenticated using end user credentials"
+)
+
 
+def get_all_projects_in_folder(folder_id=None):
```

Comment:
nit: a doctest would be handy to show what's going on here

Reply:
I'm happy to do so, but I'm not sure a doctest would work since this is making a number of calls to GCP. I'm happy to write a unit test for this with mocks if desired?